### DISCUSSION TITLE
Webmin does not set passwords for samba users?

### DISCUSSION THREAD

**Discussion by datenheim** [webmin/webmin#1851] [Q&A] [answered]
I have setup a fresh ubuntu 22.04 system, installed webmin, installed samba package from within webmin and startet to configure it (nothing unusual for me, but have longer not tried it from webmin gui).

- converted some users from unix to samba (I expect this to copy also their password, is that right)?
- created a share, nothing special
- opened firewall for all ports

Result: not able to connect from other machines, invalid passwords.

After doing manually ` sudo smbpasswd -a user1`  for all users it worked immediately.

Question: is it intended that samba users that got converted from the unix users list, have unset passwords which are not allowed by default?

**Comment by swelljoe** [comment, marked as answer]
Samba passwords are not the same hash format as Linux passwords. Webmin does not _know_ the Linux passwords because they are hashed, and, thus it is impossible for Webmin to set Samba passwords (I mean, it could set new random passwords, but it cannot do anything with hashed passwords).

Probably useful to read this: https://unix.stackexchange.com/questions/535967/possible-to-make-samba-use-etc-shadow
